### ISSUE TITLE
document installation on Linux

### ISSUE THREAD

**Issue by rinigus** [rinigus/osmscout-server#278]
compilation instructions, git clone command with all submodules, and requirements

**Comment by linuzer** [comment]
Trying to get it installed on Ubuntu 18.04, I get an error on qmake `Project ERROR: Unknown module(s) in QT: quick positioning`

Here are the steps I followed (could serve also as a first draft for an installation-steps document, only the installation of prerequisit packages is entirely missing):

```
OSMScout Server Installation:

1. install Prime Server (Prereq. for valhalla)
git clone https://github.com/kevinkreiser/prime_server.git
cd prime_server
git submodule update --init --recursive
./autogen.sh
./configure
make test -j8
sudo make install

2. install Valhalla
apt install cmake make libtool pkg-config g++ gcc jq lcov protobuf-compiler  libboost-all-dev libboost-all-dev libprotobuf-dev nodejs npm
git clone https://github.com/valhalla/valhalla.git
cd valhalla
git submodule update --init --recursive
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Release -DENABLE_NODE_BINDINGS=OFF
make -j$(nproc)
sudo make install

3. install libpostal
git clone https://github.com/openvenues/libpostal
cd libpostal
./bootstrap.sh
mkdir data
./configure --datadir=/home/tscholz/libpostal/data
make -j$(nproc)
sudo make install
sudo ldconfig

4. install geocoder-nlp
cd src
git clone https://github.com/rinigus/geocoder-nlp.git
cd geocoder/nlp
git submodule update --init --recursive
cd ..
cd ..

5. install osmscout
git clone https://github.com/rinigus/osmscout-server.git
cd osmscout-server
git submodule update --init --recursive
cd ..
ln -s ./pro/osmscout-server_console.pro ./
qmake CONFIG+="disable_mapnik disable_osmscout disable_systemd" osmscout-server_console.pro
```

**Comment by rinigus** [comment]
Good job! quick and positioning are modules and on Ubuntu, I would expect, there are separate packages providing it. Try to look through the following list that I composed as a part of porting Pure Maps to Linux Desktop:

qtdeclarative5-dev qtlocation5-dev qtmultimedia5-dev qtquickcontrols2-5-dev qtpositioning5-dev qtbase5-dev qmlscene qt5-default g++ libqt5svg5-dev python3-dev libcurl4-openssl-dev libssl-dev qml-module-qtmultimedia qml-module-qtquick2 qml-module-qtquick-controls2 qml-module-qt-labs-settings qml-module-qtlocation qml-module-qtpositioning qml-module-qtquick-window2 qml-module-qtquick-layouts

**Comment by rinigus** [comment]
As for plans, I am planning to work or help with packaging for Debian and some of its derivatives. This will happen a bit later though.

**Comment by linuzer** [comment]
Thanks for the help, I installed qtpositioning5-dev, but now I get a ton of these errors:
`QDir::mkpath: Empty or null file name
WARNING: Cannot create directory ''`

Doesn't sound like a missing package to me.

**Comment by rinigus** [comment]
So, it all compiled and runs? 

I can see that you compiled console version. This one requires config file editing and I would suggest to compile qtcontrols version (available by symlinking https://github.com/rinigus/osmscout-server/blob/master/pro/osmscout-server_qtcontrols.pro ). You are very close, please try that - it will help you with running it and, after map folder and other activities (map download) will be able to run as console application too

**Comment by linuzer** [comment]
I think I found it out: I have to run "make" after all...
This procuced a bunch of other missing dependencies, but now make ran through.

So I'm investigation now the startup and running of all...

**Comment by rinigus** [comment]
Looking at your earlier instructions : geocoder-nlp step is not needed, its pulled in as a submodule.

If make managed to go through, you should have osmscout-server in current directory. Run it with ./osmscout-server

**Comment by linuzer** [comment]
I think I got confused, make did not run through,
it stops with these final messsges:
```
src/valhallamaster.o: In Funktion »ValhallaLogger::~ValhallaLogger()«:
valhallamaster.cpp:(.text._ZN14ValhallaLoggerD2Ev[_ZN14ValhallaLoggerD5Ev]+0xf): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Logger::~Logger()«
src/valhallamaster.o: In Funktion »ValhallaLogger::~ValhallaLogger()«:
valhallamaster.cpp:(.text._ZN14ValhallaLoggerD0Ev[_ZN14ValhallaLoggerD5Ev]+0x13): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Logger::~Logger()«
src/valhallamaster.o:(.data.rel.ro._ZTI14ValhallaLogger[_ZTI14ValhallaLogger]+0x10): Warnung: undefinierter Verweis auf »typeinfo for valhalla::midgard::logging::Logger«
collect2: error: ld returned 1 exit status
Makefile:340: recipe for target 'osmscout-server' failed
make: *** [osmscout-server] Error 1
```

Thank you so much for your ongoing help!

**Comment by rinigus** [comment]
OK, I think this maybe caused by current changes in Valhalla.

One way to fix it, would be to go to Valhalla src directory, 
```
git checkout 2.7.1
```
and then reinstall it.

Alternative is to pull it from my packaging repo: [note that there could be typos, cannot check it now]

```
git clone --recursive https://github.com/rinigus/pkg-valhalla-lite.git
cd pkg-valhalla-lite
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX:PATH=/usr/local -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DENABLE_DATA_TOOLS=OFF -DENABLE_PYTHON_BINDINGS=OFF -DENABLE_SERVICES=OFF
make -j4
make install
```

and then try to compile osmscout-server again

**Comment by linuzer** [comment]
I tried now both options, the checkout of version 2.7.1 and to install it from your repo, but both is terminating with the same error that above. In between I also did a `make clean`, just to be sure. 
I googled the last message `collect2: error: ld returned 1 exit status`, but from what I found it doesn't seem to be the original error, but even reading very carefully over the whole output, I'm unable to find any other error. 

Do you still have any clue? Thank you very much for your help!
```
root@TVM2:/home/ts/osmscout-server# make -j$(nproc)
g++ -Wl,-O1 -o osmscout-server src/dbmaster.o src/main.o src/requestmapper.o src/appsettings.o src/dbmaster_route.o src/dbmaster_search.o src/dbmaster_map.o src/osmscout-server_console.o src/searchresults.o src/infohub.o src/rollinglogger.o src/consolelogger.o src/routingforhuman.o src/geomaster.o src/config.o src/mapmanager.o src/filedownloader.o src/mapmanagerfeature.o src/mapmanager_urlcollection.o src/mapmanagerfeature_packtaskworker.o src/mapnikmaster.o src/modulechecker.o src/valhallamaster.o src/mapmanager_deleterthread.o src/systemdservice.o src/util.o src/mapboxglmaster.o src/valhallamapmatcher.o src/valhallamapmatcherdbus.o src/valhallamapmatcherdbusadaptor.o src/uhttp/microhttpserver.o src/uhttp/microhttpconnectionstore.o src/uhttp/microhttpconnection.o src/uhttp/microhttpservicebase.o src/fileselector/filemodel.o src/fileselector/statfileinfo.o src/geocoder-nlp/src/postal.o src/geocoder-nlp/src/geocoder.o qrc_qml_qtcontrols.o moc_microhttpserver.o moc_filemodel.o moc_dbmaster.o moc_requestmapper.o moc_appsettings.o moc_infohub.o moc_rollinglogger.o moc_consolelogger.o moc_routingforhuman.o moc_geomaster.o moc_mapmanager.o moc_filedownloader.o moc_mapmanagerfeature.o moc_mapmanagerfeature_packtaskworker.o moc_mapnikmaster.o moc_modulechecker.o moc_valhallamaster.o moc_mapmanager_deleterthread.o moc_systemdservice.o moc_mapboxglmaster.o moc_valhallamapmatcher.o moc_valhallamapmatcherdbus.o moc_valhallamapmatcherdbusadaptor.o   -lmicrohttpd -lpostal -lmarisa -lkyotocabinet -lz -lsqlite3 -fopenmp -lQt5Quick -lQt5Qml -lQt5Network -lQt5Sql -lQt5Xml -lQt5Gui -lQt5Positioning -lQt5DBus -lQt5Core -lGL -lpthread
src/valhallamaster.o: In Funktion »ValhallaMaster::ValhallaMaster(QObject*)::{lambda(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)#1}::_FUN(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)«:
valhallamaster.cpp:(.text+0x1d): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Logger::Logger(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)«
src/valhallamaster.o: In Funktion »ValhallaMaster::callActor(ValhallaMaster::ActorType, QByteArray const&, QByteArray&)«:
valhallamaster.cpp:(.text+0x72c): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::height(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0x83c): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::route(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0x8e4): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::isochrone(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0x96c): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::locate(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0x9f4): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::matrix(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0xa7c): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::optimized_route(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0xb1c): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::trace_attributes(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0xba0): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::trace_route(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&)«
valhallamaster.cpp:(.text+0xd09): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::cleanup()«
src/valhallamaster.o: In Funktion »ValhallaMaster::start()«:
valhallamaster.cpp:(.text+0x11b1): Warnung: undefinierter Verweis auf »valhalla::tyr::actor_t::actor_t(boost::property_tree::basic_ptree<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool)«
src/valhallamaster.o: In Funktion »ValhallaMaster::ValhallaMaster(QObject*)«:
valhallamaster.cpp:(.text+0x1fad): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::RegisterLogger(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, valhalla::midgard::logging::Logger* (*)(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&))«
valhallamaster.cpp:(.text+0x208b): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Configure(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)«
src/valhallamaster.o: In Funktion »ValhallaLogger::~ValhallaLogger()«:
valhallamaster.cpp:(.text._ZN14ValhallaLoggerD2Ev[_ZN14ValhallaLoggerD5Ev]+0xf): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Logger::~Logger()«
src/valhallamaster.o: In Funktion »ValhallaLogger::~ValhallaLogger()«:
valhallamaster.cpp:(.text._ZN14ValhallaLoggerD0Ev[_ZN14ValhallaLoggerD5Ev]+0x13): Warnung: undefinierter Verweis auf »valhalla::midgard::logging::Logger::~Logger()«
src/valhallamaster.o:(.data.rel.ro._ZTI14ValhallaLogger[_ZTI14ValhallaLogger]+0x10): Warnung: undefinierter Verweis auf »typeinfo for valhalla::midgard::logging::Logger«
collect2: error: ld returned 1 exit status
Makefile:340: recipe for target 'osmscout-server' failed
make: *** [osmscout-server] Error 1
root@TVM2:/home/ts/osmscout-server#
```

**Comment by rinigus** [comment]
Morning,

your linking command (the last one) lacks valhalla library. I wonder, what does 

```
pkg-config --libs libvalhalla
```

do in your system? Possible that valhalla is not found by pkg-config. As a hack, you could add 

```
LIBS += -lvalhalla
```
 somewhere in the used PRO file of osmscout-server. After that, rerun `qmake` again and make again.
You may get some more missing libs. For me, pkg-config --libs libvalhalla returns the following which can be used as a `LIBS`:

```
LIBS += -L/usr/local/valhalla/lib64 -L/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0 -lvalhalla -lprotobuf -lboost_date_time-mt -lboost_filesystem-mt -lboost_iostreams-mt -lboost_system-mt -lcurl -lboost_regex-mt -llz4 -lz -pthread -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc
```

**Comment by linuzer** [comment]
Wow, thanks a lot, now it compiles!

Here is the line I finally used:
`LIBS += -lvalhalla -lkyotocabinet -lmarisa -lz -lsqlite3 -lprotobuf -lboost_date_time -lboost_filesystem -lboost_iostreams -lboost_syst$`

Now trying to start it up I run into the next problem: It tells me `Could not connect to any X display.
`
And true, it is only a server, I don't have any UI installed. Is it possible to run osmscout-server as a pure service without direct UI? So it only servers the API via HTTP(S)?

**Comment by rinigus** [comment]
Its possible, but complicated. cmd line interface was made mainly for my own testing and I haven't made it user-friendly. In particular, I would suggest GUI for initial configuration and map downloads/update

Can you have ssh with X11 forwarding over there? As far as I know, all OSes should have X11 servers freely available.

For CMD line interface, you would have to compile console version and read the code (starting from https://github.com/rinigus/osmscout-server/blob/master/src/main.cpp#L173) to get the options.

**Comment by linuzer** [comment]
I installeld now the ubuntu-desktop and a lot Qt-related packages and modules.
Osmscout-server starts up and displays the main window. Although I have daubts, that all required packeges are already there, because I face several issues (if I sould open seperate issues for it, please tell me...):

- On startup, I see a lot of warnings in the terminal, don't know if this is normal:
```
root@TVM2:/home/ts/osmscout-server# ./osmscout-server
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
QQmlComponent: Component is not ready
qrc:/qml/qtcontrols/StartPage.qml:39: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:37: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:46: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:56: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:85: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:98: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:108: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:132: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:159: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:168: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:174: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:196: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:205: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:211: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:247: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:264: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:270: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:281: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:282: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:44: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:78: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:85: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:101: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:109: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementLanguageSelector.qml:43: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:338: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:337: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/ElementDownloads.qml:25:8: Unable to assign [undefined] to double
qrc:/qml/qtcontrols/StartPage.qml:370: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:38: TypeError: Cannot read property of null
qrc:/qml/qtcontrols/StartPage.qml:369: TypeError: Cannot read property of null
QQmlComponent: Component is not ready
qrc:/qml/qtcontrols/StartPage.qml:370: TypeError: Cannot read property of null
QQmlComponent: Component is not ready
qrc:/qml/qtcontrols/StartPage.qml:338: TypeError: Cannot read property of null
```

- Calling "Settings" on the menu, the contents are much longer that the window can display (I have a 1920x1080 resolution), but there is no way to scroll the contents. So the lower parts are unaccessible.

- Changed settings don't get saved

- The server does not seem to be listening on neither http://127.0.0.1:8553 nor the regular IP-address.

=> So probably some basic stuff is still missing... Do you have an idea?
Thanks a lot!

**Comment by rinigus** [comment]
OK, its a half-baked interface and there are some problems, obviously. I will work on it and will provide instructions as well as packaging. It may take me 1-2 months though, as estimated now. Maybe even more.

When you start osmscout-server, please start it from the source directory. In my case, its compiled into separate dir, and I have to start it with `../build/osmscout-server` from the directory where the sources are. Its needed for now to use different `data` files.

With Settings and other windows: you should get a scroll bar if you move the mouse to the right of the window. Alternatively, mouse wheel or some other mouse related signal can be used (on touchpad, two finger scroll, for example).

In Settings, you have to specify Maps directory. See section "Maps storage" and click on Maps to select the folder. Settings should be saved when you press "Accept" on the top right.

Why do you get so many errors, I don't know, would probably have to look. But lets try to set up maps folder first and then see how far we can get

**Comment by linuzer** [comment]
> OK, its a half-baked interface...

Don't worry at all, I'm happy to work with some "hacks" to get it running!

> please start it from the source directory.

I think I do this already. In my case it is the main "osmscout-server" directory. Build puts the final binary there and I see there a lot of moc_*.cpp and *.o files. I have there also linked the ./pro/osmscout-server_qtcontrols.pro file. Or do you mean a different folder?

> you should get a scroll bar...

Unfortunateley not. Also the wheel does not seem to work. OK, I run Ubuntu in Hyper-V on a Windows-host. That could be the reason... I will check this.

> you have to specify Maps directory

I was able to do this.

>  Settings should be saved when you press "Accept"

It looks like it does not work in all places. For example unter "Settings", I cannot enable and save "Automatic activation" and "Idle timeout". Whenever I get back to the site, it is disabled again.

I chose the "Default"-profile, English, Spain and German as languages and subscribed as an example to Europe / Spain, but it tells me something about an incompatible dataset. But I'm unsure if this applies to libosmscout, as the main screen displays also such a warnig.

The main screen also states, that it has 8 worker threads and the status is idle. But trying to connect to port 8553, Firefox says "Unknown URL path"

**Comment by linuzer** [comment]
About the scroll-bar: I can wheel-scroll in any other window, like the terminal window, the Files-Application and Firefox. So I suspect, it has to do with the compilation or a missing package.

**Comment by rinigus** [comment]
Re running folder: yes, that's correct folder, let's keep that.

Re wheel and scroll bars: strange, but let's see it later

Re libosmscout: did you use https://github.com/rinigus/libosmscout or upstream? You have to use mine version for now, otherwise you can get into trouble with downloading. I hope that this is the case and by recompiling with my version all will work...

Re automatic activation: let's forget that for some time - its related to systemd and you probably don't need it for PC

Other settings sound fine (profile & languages)

**Comment by linuzer** [comment]
Re libosmscout: I didn't think to use it at all, as I understood it is optional. I thought it would use Mapbox by default. Anyway, I tried to compile your fork, but it stops also with a strange error:
```
Scanning dependencies of target WorkQueue
[ 98%] Building CXX object Tests/CMakeFiles/WorkQueue.dir/src/WorkQueue.cpp.o
/home/ts/libosmscout/Tests/src/WorkQueue.cpp: In member function ‘std::future<int> Worker::PushWork(int, int)’:
/home/ts/libosmscout/Tests/src/WorkQueue.cpp:67:41: error: ‘bind’ is not a member of ‘std’
     std::packaged_task<int()> task(std::bind(&Worker::Work,this,a,b));
                                         ^~~~
/home/ts/libosmscout/Tests/src/WorkQueue.cpp:67:41: note: suggested alternative: ‘find’
     std::packaged_task<int()> task(std::bind(&Worker::Work,this,a,b));
                                         ^~~~
                                         find
Tests/CMakeFiles/WorkQueue.dir/build.make:62: recipe for target 'Tests/CMakeFiles/WorkQueue.dir/src/WorkQueue.cpp.o' failed
make[2]: *** [Tests/CMakeFiles/WorkQueue.dir/src/WorkQueue.cpp.o] Error 1
CMakeFiles/Makefile2:2001: recipe for target 'Tests/CMakeFiles/WorkQueue.dir/all' failed
make[1]: *** [Tests/CMakeFiles/WorkQueue.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by rinigus** [comment]
You are right regarding libosmscout - it should be optional. I am traveling right now and will be able to look into it a bit later. There seem to be some strange bug when I try to reproduce the fresh install, let me look into it.

**Comment by rinigus** [comment]
OK, I have mainly issues with the network, but let's try to get then to the bottom of your errors and warnings. I wonder which version of Qt and Ubuntu do you have installed? It seems that for some reason Theme was not registered and this may have caused the errors.

As for libosmscout - we can leave it for now, it should work without

**Comment by rinigus** [comment]
I have 5.9 as well, but I run on Gentoo which makes it somewhat easier when compared to dependencies.

Theme is internal class, but there seem to be problems with its initialization (your logs). Let me check it on the weekend and see what list of dependencies is needed on ubuntu. I have 1604 on one of the machines, but maybe the packaging is similar

**Comment by linuzer** [comment]
Thank you very much for your help!
But don't worry, I'm not in an emergency situation, all this is relativly low priority. So true, I would like to get it finally up and running to investigate its capabilities, but no need for urgend actions right now.
But I do apreciate your help really a lot!

**Comment by rinigus** [comment]
I am starting to work on this, will work on writing up the instructions and retrace your steps. Should get it working soon...

**Comment by rinigus** [comment]
I wrote down installation instructions at https://github.com/rinigus/osmscout-server/blob/master/INSTALL.md . Please let me know if it works for you.

**Comment by rinigus** [comment]
The server is also available at Flathub. I am closing the issue since INSTALL.md should provide all instructions. Please reopen if something is wrong. Thank you very much for your help!

**Comment by linuzer** [comment]
Thanks a lot for your detailled description. Indeed I was able to install with this description. I had a little problem on the last make install of osmscout-server, the paths of the icons doesn't seem correct, I had to put a "../" into the Makefile ahead of the cp command.

Now even the scrolling of the pages work!

I'm investigating now the API and conducting some test (performance and handling). I will open new issues, if I have troubles.

Thank you very much again!
